src/modules/27-platform/auth-settings/pages/subscriptions/overview/CCMUsageInfo.tsx

```diff
@@ -40,14 +45,56 @@ const ActiveCloudSpend: React.FC<{
 }
 
 const CCMUsageInfo: React.FC = () => {
+  const { accountId } = useParams<AccountPathProps>()
+  const { getString } = useStrings()
   const { limitData, usageData } = useGetUsageAndLimit(ModuleName.CE)
   const { usageErrorMsg, refetchUsage, usage } = usageData
   const { limitErrorMsg, refetchLimit, limit } = limitData
+  const ccmStartTimestamp = useGetCCMTimeStamp()
+  const ccmExpiryTimestamp = useGetCCMExpiryTimeStamp()
+
+  const [annualizedSpendTimestamp, buffer] = useMemo(() => {
+    const currentTimestamp = moment()
+    const hasStartTime = moment(ccmStartTimestamp).year() >= 2015
+    if (hasStartTime) {
+      const diffInDays = currentTimestamp.diff(moment(ccmStartTimestamp), 'days')
+      const maxAllowedDate =
+        diffInDays > 365 ? moment(ccmStartTimestamp).add(Math.floor(diffInDays / 365), 'years') : ccmStartTimestamp
+      const calculationPeriod = currentTimestamp.diff(maxAllowedDate, 'days')
+      const daysBuffer = calculationPeriod > 30 ? 30 : calculationPeriod
+      return [currentTimestamp.subtract(daysBuffer, 'days').valueOf(), daysBuffer]
+    } else {
+      const ccmCurrentYearDate = moment(ccmExpiryTimestamp).year(moment().year())
+      const diffInDays = currentTimestamp.diff(ccmCurrentYearDate, 'days')
+      const daysBuffer = diffInDays < 0 || diffInDays > 30 ? 30 : diffInDays
+      return [currentTimestamp.subtract(daysBuffer, 'days').valueOf(), daysBuffer]
+    }
+  }, [ccmStartTimestamp, ccmExpiryTimestamp])
 
-  if (usageErrorMsg) {
+  const {
+    data: ccmUsageData,
+    error: ccmUsageError,
+    refetch: ccmUsageRefetch
+  } = useGetCCMLicenseUsage({
+    queryParams: {
+      accountIdentifier: accountId,
+      timestamp: annualizedSpendTimestamp
+    }
+  })
+
+  if (usageErrorMsg || ccmUsageError) {
     return (
       <ErrorContainer>
-        <PageError message={usageErrorMsg} onClick={() => refetchUsage?.()} />
+        <PageError
+          message={usageErrorMsg || ccmUsageError}
+          onClick={() => {
+            if (usageErrorMsg) {
+              refetchUsage?.()
+            } else if (ccmUsageData) {
```

Comment:
Should this be this?
```suggestion
            } else if (ccmUsageError) {
```

Reply:
yes right, replacing it with the error variable